Add android support for BlurView

cmcewen opened the pull request:
First attempt at Android support for BlurView 😄 . I also updated the example project to the latest RN version.

It depends on this library: https://github.com/500px/500px-android-blur, which provides the options for `blurRadius`, `overlayColor`, and `downsampleFactor`. 

I'm not sure what the right `rnpm` hook is to make sure it gets added to the `build.gradle` file (you can see the necessary modifications in the example), but I'm guessing you're the right folks to talk to.

The library needs a reference to an existing rendered view and then blurs on top of it - the best way I could figure out was waiting till the relevant image was loaded and then passing a ref to the native component to blur. For regular views, this could happen in `componentDidMount`. 

If the referenced view changes, right now there's no hook to re-render the blurred view (besides passing a new ref). For most people I'm guessing that's okay since it'll probably be used mostly for images, but I could give it some more thought to try and figure that piece out.

Right now for testing the `package.json` points to my fork, but obviously that will change before merge

mikelambert commented:
@Kureev , any thoughts on this? I haven't tested this functionality/patch myself yet, but I am interested in android support for react-native-blur. Let me know if I can help in any way to finish this (testing to verify it works on android, checking/updating any necessarily installation instructions, etc). Thanks!

Kureev commented on examples/Basic/package.json:
```diff
@@ -6,7 +6,8 @@
-    "react-native-blur": "^0.7.0"
+    "react": "^0.14.5",
+    "react-native": "^0.23.0",
+    "react-native-blur": "github:cmcewen/react-native-blur"
```
don't forget to add a PR to change it to official repo

grabbou replied:
I would actually suggest doing the same we did in side-menu, so that we can dynamically reference `master` w/o going through npm. Can be done in a separate PR.

Kureev commented:
Wow, you did an awesome job there! I think we should merge this asap.

сс @grabbou

grabbou commented:
Looks good, do you have any pics how does it look like?

Kureev commented:
@cmcewen can you rebase your PR and I'll merge it in?

cmcewen commented:
I'll work on it tonight! And yeah sure, happy to work on the Android side

undef1ned commented:
Any chances on this to be merged? Im using gl-react-blur but it fails on Android 4.1.2, so hopefully this will work... before I will go with own solution :)

Kureev commented:
This can and will be merged when @cmcewen rebase this branch

undef1ned commented:
@cmcewen I just tried your forked repo, however it fails on HTC Desire 500 (Android 4.1.2) with next exception:
`E/AndroidRuntime( 3145): FATAL EXCEPTION: main
E/AndroidRuntime( 3145): java.lang.NoClassDefFoundError: android.support.v8.renderscript.RenderScript
E/AndroidRuntime( 3145): 	at com.fivehundredpx.android.blur.BlurringView.initializeRenderScript(BlurringView.java:98)
E/AndroidRuntime( 3145): 	at com.fivehundredpx.android.blur.BlurringView.<init>(BlurringView.java:37)
E/AndroidRuntime( 3145): 	at com.fivehundredpx.android.blur.BlurringView.<init>(BlurringView.java:26)
E/AndroidRuntime( 3145): 	at com.cmcewen.blurview.BlurViewManager.createViewInstance(BlurViewManager.java:24)
E/AndroidRuntime( 3145): 	at com.cmcewen.blurview.BlurViewManager.createViewInstance(BlurViewManager.java:11)`

I have found this https://github.com/500px/500px-android-blur/issues/15 - it can be related

Kureev commented:
I would be glad to merge in any suitable, well-tested solution. 

Unfortunately, I don't have a lot of time for this project any more, so I can't roll it out by myself. So if you think that 500px may work, @undef1ned - compose a PR and we'll review it together 👍

cmcewen commented:
@Kureev My bad for taking so long on this - had to focus recently on getting our iOS app ready and shippped.

I rebased, updated to latest RN in the example and added some notes in the README.

Here's a picture of what it looks like:
![screen shot 2016-06-13 at 4 18 16 pm](https://cloud.githubusercontent.com/assets/1307633/16022657/f42bd7d6-3186-11e6-89a0-1d06c0be1350.png)

Kureev commented:
Thank you so much!

tuckerconnelly commented:
Omg legend, thank you 😄
